.github/templates/linux_ci_workflow.yml.in

```diff
@@ -241,7 +241,7 @@ jobs:
           # Prune all of the docker images
           docker system prune -af
   render_test_results:
-    if: always()
+    if: always() && jobs.calculate-docker-image.outcome == 'success' && jobs.build.outcome == 'success'
```

Comment:
Hmm, we explicitly want to run this job when `job.build.outcome == 'failure'`, but not when it was skipped

Reply:
i think what we want is react to the **test** job: `job.test.outcome != 'skipped'` 
if build job failed. the test outcome will guarantee to be empty right?

Reply:
alternative that @samestep suggested is to move the download artifact as the very first step of the job, and mark job as successful if no test report artifact exists. (i dunno if there's a way to mark a job as skipped, related discussion: https://gitlab.com/gitlab-org/gitlab/-/issues/15266)

Reply:
the idea would be similar to this #56071